airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -93,14 +93,12 @@ def read(
         state: Union[List[AirbyteStateMessage], MutableMapping[str, Any]] = None,
     ) -> Iterator[AirbyteMessage]:
         """Implements the Read operation from the Airbyte Specification. See https://docs.airbyte.io/architecture/airbyte-protocol."""
-        state_manager = ConnectorStateManager(state=state)
-        connector_state = state_manager.get_legacy_state()
-
         logger.info(f"Starting syncing {self.name}")
         config, internal_config = split_config(config)
         # TODO assert all streams exist in the connector
         # get the streams once in case the connector needs to make any queries to generate them
         stream_instances = {s.name: s for s in self.streams(config)}
+        state_manager = ConnectorStateManager(stream_instance_map=stream_instances, state=state)
```

Comment:
This is an interesting one. We need this stream_name -> Stream instance map as part of input to the ConnectorStateManager for a very specific edge case around namespaces. Although I don't see any tangible use case for namespaces for Python API sources at this time (no Stream's override the default `None` behavior), to not build us into a corner _if_ we ever do find one, we need this map to resolve it.

When we receive a legacy state message for the first time and are transforming it into per-stream state, we will use this map to look up the namespace and that will then be part of the descriptor of the stream state message emitted.